partner_event/models/event_registration.py

```diff
@@ -27,26 +27,34 @@ def _prepare_partner(self, vals):
             "phone": vals.get("phone", False),
         }
 
+    def _update_attendee_partner_id(self, values):
+        if not values.get("attendee_partner_id") and values.get("email"):
+            Partner = self.env["res.partner"]
+            Event = self.env["event.event"]
+            # Look for a partner with that email
+            email = values.get("email").replace("%", "").replace("_", "\\_")
+            attendee_partner = Partner.search([("email", "=ilike", email)], limit=1)
+            event = Event.browse()
+            if values.get("event_id"):
+                event = Event.browse(values["event_id"])
+            if attendee_partner:
+                for field in {"name", "phone", "mobile"}:
+                    values[field] = values.get(field) or attendee_partner[field]
+            elif event and event.create_partner:
+                # Create partner
+                attendee_partner = Partner.sudo().create(self._prepare_partner(values))
+            values["attendee_partner_id"] = attendee_partner.id
+        return values
+
     @api.model_create_multi
     def create(self, vals_list):
         for values in vals_list:
-            if not values.get("attendee_partner_id") and values.get("email"):
-                Partner = self.env["res.partner"]
-                Event = self.env["event.event"]
-                # Look for a partner with that email
-                email = values.get("email").replace("%", "").replace("_", "\\_")
-                attendee_partner = Partner.search([("email", "=ilike", email)], limit=1)
-                event = Event.browse(values["event_id"])
-                if attendee_partner:
-                    for field in {"name", "phone", "mobile"}:
-                        values[field] = values.get(field) or attendee_partner[field]
-                elif event.create_partner:
-                    # Create partner
-                    attendee_partner = Partner.sudo().create(
-                        self._prepare_partner(values)
-                    )
-                values["attendee_partner_id"] = attendee_partner.id
-        return super(EventRegistration, self).create(vals_list)
+            self._update_attendee_partner_id(values)
+        return super().create(vals_list)
+
+    def write(self, values):
```

Comment:
Use the standard signature:

```suggestion
    def write(self, vals):
```